packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			factory,
+			[[factory.type, Promise.resolve(factory)]],
+			undefined,
+			[inner],
+		);
+		const loader = createLoader(
+			[[provider.defaultCodeDetails, runtimeFactory]],
+			provider.documentServiceFactory,
+			provider.urlResolver,
+			provider.logger,
+		);
+		loaderContainerTracker.add(loader);
+		return loader.resolve({
+			url: await provider.driver.createContainerUrl(documentId, containerUrl),
+			headers,
+		});
+	}
+
+	async function createSummarizerFromContainer(container: IContainer): Promise<ISummarizer> {
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			testDataObjectFactory,
+			[[testDataObjectFactory.type, Promise.resolve(testDataObjectFactory)]],
+			undefined,
+			[innerRequestHandler],
+		);
+		const loader = createLoader(
+			[[provider.defaultCodeDetails, runtimeFactory]],
+			provider.documentServiceFactory,
+			provider.urlResolver,
+			provider.logger,
+		);
+		loaderContainerTracker.add(loader);
+		const absoluteUrl = await container.getAbsoluteUrl("");
+		if (absoluteUrl === undefined) {
+			throw new Error("URL could not be resolved");
+		}
+		const summarizer = await Summarizer.create(loader, absoluteUrl);
+		await waitForSummarizerConnection(summarizer);
+		return summarizer;
+	}
+
+	async function waitForSummarizerConnection(summarizer: ISummarizer): Promise<void> {
+		const runtime = (summarizer as any).runtime as ContainerRuntime;
+		if (!runtime.connected) {
+			return new Promise((resolve) => runtime.once("connected", () => resolve()));
+		}
+	}
+
+	it("Can load a paused container", async () => {
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+			},
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+		const initialValue = dataObject2.value;
+
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		dataObject1.increment();
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+		assert.strictEqual(
+			initialValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the initial value",
+		);
+	});
+
+	it("Can load a paused container at a specific sequence number", async () => {
+		// Min 5 ops, max 15 ops
+		const numIncrement = Math.round(Math.random() * 10) + 5;
```

Comment:
Don't use randomness in tests, since it can lead to flaky results - either use a seeded random library or select one or more specific constants to test